builder/src/lib.rs

```diff
@@ -1,10 +1,247 @@
+        if optional_fields.contains(&identstr) {
+            let ty = match &f.ty {
+                syn::Type::Path(ref path) => {
+                    match path.path.segments[0].arguments {
+                        syn::PathArguments::AngleBracketed(ref arg) => {
+                            match arg.args[0] {
+                                syn::GenericArgument::Type(ref t) => t.clone(),
+                                _ => unreachable!(),
+                            }
+                        },
+                        _ => unreachable!(),
+                    }
+                },
+                _ => unreachable!(),
+            };
+            quote! {
+                pub fn #ident(&mut self, #ident: #ty) -> &mut Self {
+                    self.#ident = std::option::Option::Some(#ident);
+                    self
+                }
+            }
+        } else if builders.contains_key(&identstr) {
+            // this should be def_site, but I'm unable to get the config to enable that method
+            let methodname = Ident::new(&builders.get(&identstr).unwrap(), proc_macro2::Span::call_site());
+            let ty = match &f.ty {
+                syn::Type::Path(ref path) => {
+                    match path.path.segments[0].arguments {
+                        syn::PathArguments::AngleBracketed(ref arg) => {
+                            match arg.args[0] {
+                                syn::GenericArgument::Type(ref t) => t.clone(),
+                                _ => unreachable!(),
+                            }
+                        },
+                        _ => unreachable!(),
+                    }
+                },
+                _ => unreachable!(),
+            };
+            quote! {
+                pub fn #methodname(&mut self, item: #ty) -> &mut Self {
+                    self.#ident.push(item);
+                    self
+                }
+            }
+        } else {
+            quote! {
+                pub fn #ident(&mut self, #ident: #ty) -> &mut Self {
+                    self.#ident = std::option::Option::Some(#ident);
+                    self
+                }
+            }
+        }
+    });
+    quote! { #(#setters)* }
+}
+
+fn expand_build(name: &Ident, fields: &syn::FieldsNamed, optional_fields: &HashSet<String>, builders: &HashMap<String, String>) -> TokenStream2 {
+    let validation = fields.named.iter().filter(|f| {
+        let identstr = f.ident.as_ref().map(|x| format!("{}", x)).unwrap_or("".to_owned());
+        !optional_fields.contains(&identstr) && !builders.contains_key(&identstr)
+    }).map(|f| {
+        let ident = &f.ident;
+        let identstr = f.ident.as_ref().map(|x| format!("{}", x)).unwrap_or("".to_owned());
+        quote!{
+            if self.#ident.is_none() {
+                return Err(<std::boxed::Box<dyn std::error::Error>>::from(format!("{} is missing", #identstr)));
```

Comment:
I found a more concise way to write this -- since the compiler can infer the exact error type here based on what is declared in the function signature.

```rust
return Err(<_>::from(format!("{} is missing", #identstr)));
```